docs/DeveloperGuide.md

```diff
@@ -6,8 +6,77 @@
+- **Ui** manages all output and user interaction, such as printing data and messages to the console.
+- **TrainingSession** represents a single training session, including exercises and metadata (like date and description).
+- **Exercise** represents different types of exercises available in the application, like pull-ups or shuttle runs.
+
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Edit Exercise Feature
+
+The **Edit Exercise** feature is managed by the `TrainingSession` class, and is primarily carried out by its 
+`editExercise()` function. This feature utilizes the `setPerformance()`and `getReps()` methods from the 
+`ExerciseStation` classes to edit the repetitions and timings for the user’s selected 
+exercises. Additionally, it calculates the points the user will earn for each exercise based on the updated values.
+
+#### Step 1: Logging a New Training Session
+
+When the user logs a new training session, an instance of the `TrainingSession` class is created. This instance 
+initializes an EnumMap, which populates the various `ExerciseStation` instances with their initial values. 
+Below is a representation of how the `ExerciseStations` are initialized:
+
+![TrainingSessionInitialState.png](TrainingSessionInitialState.png)  
+
+#### Step 2: Editing a Training Session
+
+When the user wishes to edit a training session, they specify an `Exercise` Enum, and the reps/timing to be inputted,
+which is passed to the`editExercise` function. This function calls the relevant methods to update the repetitions or 
+timings and calculates the corresponding points for the specified exercise.
+
+The following sequence diagram illustrates the function calls involved in this process:
+
+![editExerciseSequenceDiagram.png](editExerciseSequenceDiagram.png)
+
+Additionally, the state diagram below shows the end state of the `editExercise` function after execution of the command,
+`editExercise(Exercise.PULL_UP, 1)`:
+
+![TrainingSessionEditState.png](TrainingSessionEditState.png)
+
+### Points Calculation Feature
+
+The **points calculation feature** is a significant part of the `ExerciseStation` system. It allows for the calculation
+of user-specific points based on their performance in various exercises (e.g., pull-ups, sit-ups). This process 
+involves interaction between the `ExerciseStation` and the `Calculator` classes, ensuring that the correct points are
+assigned based on predefined lookup tables.
+
+#### 1. Class Interaction Overview
+
+Each `ExerciseStation` subclass (e.g., `PullUpStation`, `SitUpStation`) has its own implementation of the `getPoints()` 
+method. The main responsibility of this method is to invoke the `calculatePoints()` function from the 
+respective **calculator** class (e.g., `PullUpCalculator`, `SitUpCalculator`), which holds the points calculation logic.
+
+![getPointsSequenceDiagram.png](getPointsSequenceDiagram.png)
```

Comment:
Could the "PullUpStation" and "PullUpCalculator" be named to be generic names to showcase what is written in the description?